packages/@ember/-internals/glimmer/index.ts

```diff
@@ -142,6 +142,157 @@
+    End date: {{range.end}}
+  </DateRanges>
+  ```
+
+  ```app/components/date-ranges.hbs
+  <div>
+    {{yield (hash start=@value.start end=@value.end)}}
+  </div>
+  ```
+
+  Result:
+
+  ```html
+  <div>
+    Start Date: July 1st
+    End Date: July 30th
+  </div>
+  ```
+
+  Multiple values can be yielded as block params:
+
+  ```app/templates/application.hbs
+  <Banner @value={{@model}} as |title subtitle body|>
+    <h1>{{title}}</h1>
+    <h2>{{subtitle}}</h2>
+    {{body}}
+  </Banner>
+  ```
+
+  ```app/components/Banner.hbs
+  <div>
+    {{yield "Hello title" "hello subtitle" "body text"}}
+  </div>
+  ```
+
+  Result:
+
+  ```html
+  <div>
+    <h1>Hello title</h1>
+    <h2>hello subtitle</h2>
+    body text
+  </div>
+  ```
+
+  However, it is preferred to use the hash helper, as this can prevent breaking changes to your component and also simplify the api for the component.
+
+  Multiple components can be yielded with the `hash` and `component` helper:
+
+  ```app/templates/application.hbs
+  <Banner @value={{@model}} as |banner|>
+    <banner.Title>Banner title</banner.Title>
+    <banner.Subtitle>Banner subtitle</banner.Subtitle>
+    <banner.Body>A load of body text</banner.Body>
+  </Banner>
+  ```
+
+  ```app/components/banner.ts
+  import Title from './title';
+  import Subtitle from './subtitle';
+  import Body from './body';
+
+  export default class Banner extends Component {
+    Title = Title;
+    Subtitle = Subtitle;
+    Body = Body;
+  }
+  ```
+
+  ```app/components/banner.hbs
+  <div>
+    {{yield (hash
+      Title=this.Title
+      Subtitle=this.Subtitle
+      Body=(component this.Body defaultArg="some value")
+    )}}
+  </div>
+  ```
+
+  Result:
+
+  ```html
+  <div>
+    <h1>Banner title</h1>
+    <h2>Banner subtitle</h2>
+    A load of body text
+  </div>
+  ```
+
+  A benefit of using this pattern is that the user of the component can change the order the components are displayed.
+
+  ```app/templates/application.hbs
+  <Banner @value={{@model}} as |banner|>
+    <banner.Subtitle>Banner subtitle</banner.Subtitle>
+    <banner.Title>Banner title</banner.Title>
+    <banner.Body>A load of body text</banner.Body>
+  </Banner>
+  ```
+
+  Result:
+
+  ```html
+  <div>
+    <h2>Banner subtitle</h2>
+    <h1>Banner title</h1>
+    A load of body text
+  </div>
+  ```
+
+  Another benefit to using `yield` with the `hash` and `component` helper
+  is you can pass attributes and arguments to these components:
+
+  ```app/templates/application.hbs
+  <Banner @value={{@model}} as |banner|>
+    <banner.Subtitle class="mb-1">Banner subtitle</banner.Subtitle>
+    <banner.Title @variant="loud">Banner title</banner.Title>
+    <banner.Body>A load of body text</banner.Body>
+  </Banner>
+  ```
+
+  ```app/components/banner/subtitle.hbs
+  {{!-- note the use of ..attributes --}}
+  <div ...attributes>
+    {{yield}}
+  </div>
+  ```
+
+  ```app/components/banner/title.hbs
+  {{#if @variant "loud"}}
```

Comment:
```suggestion
  {{#if (eq @variant "loud")}}
```

The current example throws a syntax error.

Reply:
Thanks for catching this!

Reply:
Truth helpers haven't been shipped yet, have they? Using a helper, which is not available by default, might be confusing.

Reply:
Let's get them shipped 🙃 

Cc @chancancode 🥳